### PR TITLE
Bump open-aea to v1.12.0

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Bump open-aea to latest

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dvilelaf — COMMENTED** [review]
Looks good, although we have several failing tests. Are those related to the version bump?

**Comment by angrybayblade** [comment]
So 4 Failures in total.

1. `FAILED tests/test_autonomy/test_cli/test_replay/test_tendermint.py::TestTendermintRunner::test_run` caused by

```
E         [Tendermint] Successfully initialized 4 node directories
E         Error: Cannot find valory/open-autonomy-tendermint:0.1.0, Please build images first.
```

2. On macOS network failure while installing dependencies, again !

3. Same network failure on windows

4. E2E

```
FAILED tests/test_agents/test_oracle_abci.py::TestAgentCatchup::test_run[4]
FAILED tests/test_agents/test_register_reset_abci.py::TestTendermintResetInterruptNoRejoin::test_run[4]
FAILED tests/test_agents/test_register_reset_abci.py::TestTendermintReset::test_run[4]
FAILED tests/test_agents/test_register_reset_abci.py::TestTendermintResetInterrupt::test_run[4]
```

I found this while going through the logs

```
err="precommit step; +2/3 prevoted for an invalid block: wrong Block.Header.AppHash.  Expected 726F6F743A3072657365743A30, got 726F6F743A3172657365743A31" stack="goroutine 24972795 [running]:\nruntime/debug.Stack()\n\t/opt/hostedtoolcache/go/1.18.0/x64/src/runtime/debug/stack.go:24 +0x65\ngithub.com/tendermint/tendermint/consensus.(*State).receiveRoutine.func2()\n\t/home/runner/work/tendermint/tendermint/consensus/state.go:727 +0x4c\npanic({0xe554c0, 0xc001dfb460})\n\t/opt/hostedtoolcache/go/1.18.0/x64/src/runtime/panic.go:838 +0x207\ngithub.com/tendermint/tendermint/consensus.(*State).enterPrecommit(0xc00057a700, 0x11, 0x0)\n\t/home/runner/work/tendermint/tendermint/consensus/state.go:1408 +0x16ff\ngithub.com/tendermint/tendermint/consensus.(*State).addVote(0xc00057a700, 0xc001e875e0, {0xc000d40b10, 0x28})\n\t/home/runner/work/tendermint/tendermint/consensus/state.go:2123 +0x188f\ngithub.com/tendermint/tendermint/consensus.(*State).tryAddVote(0xc00057a700, 0xc001e875e0, {0xc000d40b10?, 0xc00055b700?})\n\t/home/runner/work/tendermint/tendermint/consensus/state.go:1949 +0x2c\ngithub.com/tendermint/tendermint/consensus.(*State).handleMsg(0xc00057a700, {{0x12901e0?, 0xc00000f8f8?}, {0xc000d40b10?, 0x0?}})\n\t/home/runner/work/tendermint/tendermint/consensus/state.go:856 +0x44b\ngithub.com/tendermint/tendermint/consensus.(*State).receiveRoutine(0xc00057a700, 0x0)\n\t/home/runner/work/tendermint/tendermint/consensus/state.go:763 +0x419\ncreated by github.com/tendermint/tendermint/consensus.(*State).OnStart\n\t/home/runner/work/tendermint/tendermint/consensus/state.go:379 +0x12d\n"
```

**Comment by dvilelaf** [comment]
What's up with `tests/test_autonomy/test_cli/test_replay/test_tendermint.py::TestTendermintRunner::test_run` in Ubuntu + py3.10?

File: tests/test_autonomy/test_deploy/test_service_specification.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+        agents = spec.generate_agents()
+        assert len(agents) == 1, agents
+
+        agent = spec.generate_agent(0)
+        assert len(agent.keys()) == 19, agent
+
+        spec.service.overrides = []
+        agent = spec.generate_agent(0)
+        assert len(agent.keys()) == 6, agent
```
Comment (Karrenbelt) [inline tests/test_autonomy/test_deploy/test_service_specification.py:107]: these numbers ideally also as constants so we know what they represent

File: tests/test_autonomy/test_deploy/test_service_specification.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+        self.keys_path.write_text(
+            json.dumps(
+                [
+                    {
+                        "address": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+                    }
+                ]
+            )
+        )
```
Comment (Karrenbelt) [inline tests/test_autonomy/test_deploy/test_service_specification.py:141]:
```
address = "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
self.keys_path.write_text(json.dumps([dict(address=address)]))
```

**Review by Karrenbelt — APPROVED** [review]
LGTM

e2e failures are know culprits:
```
2022-07-08T14:38:45.3821659Z =========================== short test summary info ============================
2022-07-08T14:38:45.3821860Z FAILED tests/test_agents/test_register_reset_abci.py::TestTendermintReset::test_run[4]
2022-07-08T14:38:45.3822071Z FAILED tests/test_agents/test_register_reset_abci.py::TestTendermintResetInterrupt::test_run[4]
2022-07-08T14:38:45.3822236Z = 2 failed, 18 passed, 8 skipped, 1298 deselected, 1 warning in 4229.89s (1:10:29) =
```

**Pull request merged**